src/services/statistic/index.js

```diff
@@ -14,141 +14,72 @@ const lessonModel = require('../lesson/model');
 	return userModel.userModel.findById(userId).populate('roles').then((user) => {
 		user.roles = Array.from(user.roles);
 			if (user.roles.filter(u => (u.name === 'administrator')).length) {
 				return [
 					{
 						name: 'users',
-						promise: userModel.userModel.count().exec(),
-						model: userModel.userModel.find().exec()
+						promise: userModel.userModel.count({schoolId: user.schoolId}).exec(),
+						model: userModel.userModel.find({schoolId: user.schoolId}).exec()
 					},
-					{
+					/*{
 						name: 'accounts',
 						promise: accountModel.count().exec(),
 						model: accountModel.find().exec()
-					},
+					},*/
 					{
 						name: 'homework',
-						promise: homeworkModel.homeworkModel.count().exec(),
-						model: homeworkModel.homeworkModel.find().exec()
+						promise: homeworkModel.homeworkModel.count({schoolId: user.schoolId}).exec(),
+						model: homeworkModel.homeworkModel.find({schoolId: user.schoolId}).exec()
 					},
 					{
 						name: 'submissions',
-						promise: homeworkModel.submissionModel.count().exec(),
-						model: homeworkModel.submissionModel.find().exec()
+						promise: homeworkModel.submissionModel.count({schoolId: user.schoolId}).exec(),
+						model: homeworkModel.submissionModel.find({schoolId: user.schoolId}).exec()
 					},
-					{
+					/*{
 						name: 'comments',
 						promise: homeworkModel.commentModel.count().exec(),
 						model: homeworkModel.commentModel.find().exec()
-					},
-					{
+					},*/
+					/*{
 						name: 'lessons',
 						promise: lessonModel.count().exec(),
 						model: lessonModel.find().exec()
-					},
+					},*/
 					{
 						name: 'classes',
-						promise: groupModel.classModel.count().exec(),
-						model: groupModel.classModel.find().exec()
+						promise: groupModel.classModel.count({schoolId: user.schoolId}).exec(),
+						model: groupModel.classModel.find({schoolId: user.schoolId}).exec()
 					},
 					{
 						name: 'courses',
-						promise: groupModel.courseModel.count().exec(),
-						model: groupModel.courseModel.find().exec()
+						promise: groupModel.courseModel.count({schoolId: user.schoolId}).exec(),
+						model: groupModel.courseModel.find({schoolId: user.schoolId}).exec()
 					},
 					{
 						name: 'teachers',
-						promise: userModel.userModel.count({roles: '0000d186816abba584714c98'}).exec(),
-						model: userModel.userModel.find({roles: '0000d186816abba584714c98'}).exec()
+						promise: userModel.userModel.count({roles: '0000d186816abba584714c98', schoolId: user.schoolId}).exec(),
+						model: userModel.userModel.find({roles: '0000d186816abba584714c98',schoolId: user.schoolId}).exec()
 					},
 					{
 						name: 'students',
-						promise: userModel.userModel.count({roles: '0000d186816abba584714c99'}).exec(),
-						model: userModel.userModel.find({roles: '0000d186816abba584714c99'}).exec()
+						promise: userModel.userModel.count({roles: '0000d186816abba584714c99', schoolId: user.schoolId}).exec(),
```

Comment:
const roleService   = this.app.service('/roles');
roleService.find({})
For take roleIds directly from DB.

Reply:
Do you think in this context this works like this:

`roles: this.app.service('/roles').find({name: "student"}).then(role=> (return role._id))`

in this context. It was used like this from the beginning of the statistics. But I get your concerns.

Reply:
```	
Yes, at start is a good point and use query $or to select.

const selectingRoles= ['student','teacher'].reduce((arr,roleName) => {
	arr.push({'name':roleName});
	return arr
},[]);

roleService.find({query:{$or:selectingRoles}}).then(roles=>{
	//take the ids
})
```